nntrainer/layers/loss/kld_loss_layer.cpp

```diff
@@ -20,9 +21,50 @@ KLDLossLayer::KLDLossLayer() {}
 
 KLDLossLayer::~KLDLossLayer() {}
 
-void KLDLossLayer::setProperty(const std::vector<std::string> &values) {}
+void KLDLossLayer::setProperty(const std::vector<std::string> &values) {
+  if (values.size()) {
+    throw std::invalid_argument(
+      "kld loss does not take any properties, but values given");
+  }
+}
 
-void KLDLossLayer::forwarding(RunLayerContext &context, bool training) {}
+void KLDLossLayer::forwarding(RunLayerContext &context, bool training) {
+  // -0.5 * sum(1 + log_std - pow(mu, 2) - exp(log_std))
+  auto &mu = context.getInput(0);
+  auto &log_std = context.getInput(1);
+  auto &ret = context.getOutput(0);
+  auto &temp = context.getTensor(temp_idx);
+  auto &before_sum = context.getTensor(before_sum_idx);
 
-void KLDLossLayer::calcDerivative(RunLayerContext &context) {}
+  mu.pow(2.0f, temp);                 // 1. temp = mu ^ 2
+  log_std.subtract(temp, before_sum); // 2. before_sum = log_std - temp
+  log_std.apply<float>(expf, temp);   // 3. temp = exp(log_std) - 1
+  temp.subtract_i(1.0f);
+  before_sum.subtract_i(temp);          // 4. before_sum = before_sum - temp
+  before_sum.sum({1, 2, 3}, ret, -0.5); // 5. sum * 0.5
```

Comment:
I don't fully understand the forward implementation of KLD Loss in commit 3. 
Are the written formulas and actual implementations the same?

I would like to know the difference between the formula on line 32 of kld_loss_layer.cpp and the PyTorch's KLD formula added to the PR.

In PyTorch, it is summarized as follows:
```
if not log_target: # default
    loss_pointwise = target * (target.log() - input)
else:
    loss_pointwise = target.exp() * (target - input)
```